File: examples/java-maven-example/src/test/java/org/galaxio/performance/picatinny/scenarios/PicatinnyScenario.java
package org.galaxio.performance.picatinny.scenarios;

import io.gatling.javaapi.core.ScenarioBuilder;
import org.galaxio.performance.picatinny.cases.PicatinnyCases;
import org.galaxio.performance.picatinny.feeders.GeneratedPicatinnyFeeders;
import org.galaxio.performance.picatinny.feeders.PicatinnyFeeders;

import java.util.Iterator;
import java.util.Map;

import static io.gatling.javaapi.core.CoreDsl.scenario;

public final class PicatinnyScenario {
    private PicatinnyScenario() {
    }

    public static ScenarioBuilder apply(String name, String transactionName) {
        return withFeeders(name).exec(PicatinnyCases.businessOperation(transactionName));
    }

    public static ScenarioBuilder apply(String name) {
        return withFeeders(name).exec(PicatinnyCases.scenarioOperation());
    }

    static ScenarioBuilder withFeeders(String name) {
        ScenarioBuilder builder = scenario(name);
        for (Iterator<Map<String, Object>> feeder : PicatinnyFeeders.legacy()) {
            builder = builder.feed(feeder);
        }
        builder = builder
                .feed(GeneratedPicatinnyFeeders.generatedUsers())
                .feed(GeneratedPicatinnyFeeders.governmentIds())
                .feed(GeneratedPicatinnyFeeders.finance());
        return builder;
    }
}

File: examples/kotlin-gradle-example/src/gatling/kotlin/org/galaxio/performance/picatinny/scenarios/PicatinnyScenario.kt
package org.galaxio.performance.picatinny.scenarios

import io.gatling.javaapi.core.CoreDsl.scenario
import io.gatling.javaapi.core.ScenarioBuilder
import org.galaxio.performance.picatinny.cases.PicatinnyCases
import org.galaxio.performance.picatinny.feeders.GeneratedPicatinnyFeeders
import org.galaxio.performance.picatinny.feeders.PicatinnyFeeders

object PicatinnyScenario {
    fun apply(name: String, transactionName: String): ScenarioBuilder =
        withFeeders(name).exec(PicatinnyCases.businessOperation(transactionName))

    fun apply(name: String): ScenarioBuilder =
        withFeeders(name).exec(PicatinnyCases.scenarioOperation())

    internal fun withFeeders(name: String): ScenarioBuilder {
        var builder = scenario(name)
        PicatinnyFeeders.legacy().forEach { feeder ->
            builder = builder.feed(feeder)
        }
        builder = builder
            .feed(GeneratedPicatinnyFeeders.generatedUsers())
            .feed(GeneratedPicatinnyFeeders.governmentIds())
            .feed(GeneratedPicatinnyFeeders.finance())
        return builder
    }
}
